### PR TITLE
fix: Add level and logger name to default log format

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes https://github.com/meltano/sdk/issues/1232

----
:books: Documentation preview :books:: https://meltano-sdk--1233.org.readthedocs.build/en/1233/

### CODE REVIEW

**Review by aaronsteers — APPROVED** [review]

File: singer_sdk/default_logging.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: 1
 disable_existing_loggers: false
 formatters:
   console:
-    format: "{asctime} {message}"
+    format: "{asctime} {levelname} {name} {message}"
```
Comment (aaronsteers) [inline singer_sdk/default_logging.yml:5]: @edgarrmondragon - Do we need logger name? Also, what do you think of a colon or other delimiter prior to {message}?

**Review by aaronsteers — APPROVED** [review]

File: singer_sdk/default_logging.yml (hunk 1 above)
Comment (aaronsteers) [reply singer_sdk/default_logging.yml:5]: (Non-blocking from my side. You can make the call if no concerns from others.)

Comment (edgarrmondragon) [reply singer_sdk/default_logging.yml:5]: <details>
<summary>Current output:</summary>

```
2022-12-05 19:45:11,689 INFO my_tap Added 'child' as child stream to 'my_stream'
2022-12-05 19:45:11,690 INFO my_tap Beginning incremental sync of 'my_stream'...
2022-12-05 19:45:11,690 INFO my_tap Tap has custom mapper. Using 1 provided map(s).
2022-12-05 19:45:11,690 INFO my_tap Beginning full_table sync of 'child' with context: {'parent_id': 1}...
2022-12-05 19:45:11,690 INFO my_tap Tap has custom mapper. Using 1 provided map(s).
2022-12-05 19:45:11,692 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.0004761219024658203, "tags": {"stream": "child", "context": {"parent_id": 1}, "status": "succeeded"}}
2022-12-05 19:45:11,692 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "child", "context": {"parent_id": 1}}}
2022-12-05 19:45:11,693 INFO my_tap Beginning full_table sync of 'child' with context: {'parent_id': 2}...
2022-12-05 19:45:11,693 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.0004048347473144531, "tags": {"stream": "child", "context": {"parent_id": 2}, "status": "succeeded"}}
2022-12-05 19:45:11,693 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "child", "context": {"parent_id": 2}}}
2022-12-05 19:45:11,694 INFO my_tap Beginning full_table sync of 'child' with context: {'parent_id': 3}...
2022-12-05 19:45:11,695 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.000492095947265625, "tags": {"stream": "child", "context": {"parent_id": 3}, "status": "succeeded"}}
2022-12-05 19:45:11,695 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "child", "context": {"parent_id": 3}}}
2022-12-05 19:45:11,695 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.004832029342651367, "tags": {"stream": "my_stream", "context": {}, "status": "succeeded"}}
2022-12-05 19:45:11,695 INFO singer_sdk.metrics INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "my_stream", "context": {}}}
```

</details>

@aaronsteers wdyt of a pipe-delimited format

```yaml
formatters:
  console:
    format: "{asctime:23s} | {levelname:8s} | {name:20s} | {message}"
    style: "{"
```

<details>
<summary>Pipe-delimited:</summary>

```
2022-12-05 19:46:46,744 | INFO     | my_tap               | Added 'child' as child stream to 'my_stream'
2022-12-05 19:46:46,744 | INFO     | my_tap               | Beginning incremental sync of 'my_stream'...
2022-12-05 19:46:46,744 | INFO     | my_tap               | Tap has custom mapper. Using 1 provided map(s).
2022-12-05 19:46:46,745 | INFO     | my_tap               | Beginning full_table sync of 'child' with context: {'parent_id': 1}...
2022-12-05 19:46:46,745 | INFO     | my_tap               | Tap has custom mapper. Using 1 provided map(s).
2022-12-05 19:46:46,746 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.0005319118499755859, "tags": {"stream": "child", "context": {"parent_id": 1}, "status": "succeeded"}}
2022-12-05 19:46:46,747 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "child", "context": {"parent_id": 1}}}
2022-12-05 19:46:46,747 | INFO     | my_tap               | Beginning full_table sync of 'child' with context: {'parent_id': 2}...
2022-12-05 19:46:46,748 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.0004410743713378906, "tags": {"stream": "child", "context": {"parent_id": 2}, "status": "succeeded"}}
2022-12-05 19:46:46,748 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "child", "context": {"parent_id": 2}}}
2022-12-05 19:46:46,749 | INFO     | my_tap               | Beginning full_table sync of 'child' with context: {'parent_id': 3}...
2022-12-05 19:46:46,749 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.0004508495330810547, "tags": {"stream": "child", "context": {"parent_id": 3}, "status": "succeeded"}}
2022-12-05 19:46:46,750 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "child", "context": {"parent_id": 3}}}
2022-12-05 19:46:46,750 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "timer", "metric": "sync_duration", "value": 0.0052759647369384766, "tags": {"stream": "my_stream", "context": {}, "status": "succeeded"}}
2022-12-05 19:46:46,750 | INFO     | singer_sdk.metrics   | INFO METRIC: {"metric_type": "counter", "metric": "record_count", "value": 3, "tags": {"stream": "my_stream", "context": {}}}
```

</details>

Comment (edgarrmondragon) [reply singer_sdk/default_logging.yml:5]: (That's more human-readable imho) and can users can still override it with a logging config file

**Pull request merged**